frontend/src/host_orchestrator/orchestrator/operation.go

```diff
@@ -44,18 +45,19 @@ type OperationResultError struct {
 type OperationManager interface {
 	New() Operation
 
-	Get(string) (Operation, error)
+	Get(name string) (Operation, error)
 
-	Complete(string, OperationResult) error
+	Complete(name string, result OperationResult) error
 
-	// TODO(b/231319087) This should handle timeout.
-	Wait(string) (Operation, error)
+	// Waits for the specified operation to be DONE within the passed deadline. If the deadline
+	// is reached it will return the current state of the operation which might be DONE or not.
+	Wait(name string, dt time.Duration) (Operation, error)
 }
 
 type mapOMOperationEntry struct {
-	data      Operation
-	mutex     sync.RWMutex
-	waitGroup sync.WaitGroup
+	data  Operation
+	mutex sync.RWMutex
+	done  chan struct{}
```

Comment:
Using a channel here suggests information is going to be sent through it, but whatever that is it's ignored on line 134. Also, not sure if this applies, but using a channel means the operation can only be waited on as many times as the channel is written into (usually once).
Maybe it's better to use a conditional variable for this (`sync.Cond`)

Reply:
After a channel has been closed (and drained of all sent values, not relevant here), subsequent receive operations proceed immediately, yielding zero values. This capability  of channels is common practice in Golang to create a broadcast mechanism: don't send a value on the channel, just close it.